rust/operator-binary/src/kerberos.rs

```diff
@@ -147,11 +154,11 @@ impl CoreSiteConfigBuilder {
 /// ```
 ///
 /// After we have switched to using the following principals everything worked without problems
-fn principal_host_part(hdfs: &HdfsCluster) -> Result<String, Error> {
+fn principal_host_part(hdfs: &HdfsCluster) -> Result<String> {
     let hdfs_name = hdfs.name_any();
     let hdfs_namespace = hdfs
         .namespace_or_error()
-        .map_err(|_| Error::ObjectHasNoNamespace {
+        .with_context(|_| ObjectHasNoNamespaceSnafu {
```

Comment:
question: Why can't we use `.context()` here? No complaint, just me asking for a short explanation.

Reply:
Ah, good point - didn't cross my mind. Thank you for the explanation!